src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  const adder = (a) => {
+    if (a !== undefined) {
+      amount += a;
+
+      return adder;
+    } else {
+      const result = amount;
+
+      amount = 0;
+
+      return result;
```

Comment:
Consider renaming the parameter `a` to something more descriptive, such as `numberToAdd`. Descriptive names make the code more readable and maintainable.